.github/workflows/pr-realm-js.yml

```diff
@@ -0,0 +1,340 @@
+        if: ${{ runner.os != 'Windows' }}
+        with:
+          key: ${{ runner.os }}-${{ matrix.variant.os }}-${{ matrix.variant.arch }}
+
+      - name: Prepend ccache executables to the PATH
+        if: ${{ runner.os != 'Windows' }}
+        run: |
+          echo "/usr/lib/ccache" >> $GITHUB_PATH
+          echo "/usr/local/opt/ccache/libexec" >> $GITHUB_PATH
+
+      - name: Install deps
+        run: npm ci --ignore-scripts
+
+      # Install node dependencies
+      - name: Install prebuilds
+        if: steps.cache-prebuild.outputs.cache-hit == 'true'
+        run: npm ci
+
+      # build the c++ library for standard targets
+      - name: Build node
+        if: ${{ (matrix.variant.os != 'ios') && (matrix.variant.os != 'android') }}
+        run: npm run prebuild:node -- --arch=${{matrix.variant.arch}}
+
+      # build the c++ library for IOS
+      - name: Build IOS
+        if: ${{ (matrix.variant.os == 'ios') }}
+        run: npm run prebuild:ios
+
+      # build the c++ library for Android
+      - name: Build Android
+        if: ${{ (matrix.variant.os == 'android') }}
+        run: npm run prebuild:android -- --arch=${{matrix.variant.arch}}
+
+      - name: Install the node dependencies in the test folder
+        working-directory: ./tests
+        run: npm ci --ignore-scripts
+
+      # all tests passed, upload artifacts
+      # the dummy file included is to force preserving paths
+      - name: Upload prebuild artifact
+        uses: actions/upload-artifact@v2
+        with:
+          name: realm-js-prebuilds
+          path: |
+            dependencies.list
+            ${{ matrix.variant.artifact-path }}
+
+  publish:
+    runs-on: ubuntu-latest
+    needs: [build-and-test]
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+
+      - name: Setup node version
+        uses: actions/setup-node@v2
+        with:
+          node-version: '14'
+          cache: 'npm'
+          cache-dependency-path: '**/package-lock.json'
+
+      - name: Setup NPM version
+        run: npm install -g npm@7
+
+      - name: Get NPM cache directory
+        id: npm-cache-dir
+        run: echo "::set-output name=dir::$(npm config get cache)"
+
+      - name: Restore NPM cache
+        id: npm-cache
+        uses: actions/cache@v2
+        with:
+          path: ${{ steps.npm-cache-dir.outputs.dir }}
+          key: ${{ runner.os }}-node-${{ hashFiles('**/package-lock.json') }}
+          restore-keys: |
+            ${{ runner.os }}-node-
+
+      - name: Install node modules
+        run: npm ci --ignore-scripts
+
+      - name: Download prebuilds
+        uses: actions/download-artifact@v2
+        with:
+          name: realm-js-prebuilds
+
+      - name: Display structure of downloaded files
+        run: ls -R prebuilds react-native/android/src/main/jniLibs
+
+      - name: Publish to Github
+        run: npx prebuild --upload-all ${{ secrets.GITHUB_TOKEN }}
```

Comment:
Should it be publishing here? This action creates a tag and a release. Seems a lot for what's just a PR at this point.

When I'm testing the release workflow I have to remove those first, otherwise it fails because the release already exists.

Reply:
@amtep sorry for the late reply.  I'm am now getting this all under review and would like to wrap everything up and get it merged ASAP.
The publish step is super old and was just testing the capabilities of publishing to github.  This should be removed, as we are hosting those on AWS.  Also, publishing should only happen on release, unless we start supplying nightly builds for users to test.

Reply:
Yes.  There is a PR In the works for the publishing part.

Reply:
[Off-topic]  Adding a `nightly` or `canary` flow would actually help a lot when we're debugging issues with our users -- so we won't be asking them to wait for a new official release.  Especially for those cases where we don't have reproductions, but have a notion that we may have fixed the issue.